docs/flow/reference/components/discovery.kubernetes.md

```diff
@@ -0,0 +1,163 @@
+
+If you supply no connection information, this component will default to using an in-cluster config. Otherwise, you can specify a kubeconfig file, or manual apiserver configuration.
+
+## Example
+
+Simple in-cluster discovery of all pods:
+
+```river
+discovery.kubernetes "k8s_pods" {
+  role = "pod"
+}
+```
+
+Specific namespace only with kubeconfig file:
+
+```river
+discovery.kubernetes "k8s_pods" {
+  role = "pod"
+  kubeconfig_file = "/path/to/kubeconfig"
+  namepsaces {
+    names = ["myapp"]
+  }
+}
+```
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`api_server` | `string` | URL of kubernetes API server | | no
+`role` | `string` | Type of kubernetes resource to query. Must be one of `node`, `pod`, `service`, `endpoints`, `endpointslice` or `ingress` | | **yes**
+`kubeconfig_file` | `string` | Path of kubeconfig file to use for kubernetes connection | | no
+
+The following subblocks are supported:
+
+Name | Description | Required
+---- | ----------- | --------
+[`namespaces`](#namespaces-block) | Information about which kubernetes namespaces to search. | no
+[`selectors`](#selectors-block) | Selectors to limit objects selected | no
+[`http_client_config`](#http_client_config-block) | HTTP client configuration for kubernetes requests | no
+
+### `namespaces` block
+
+Optional namespace discovery. If omitted, all namespaces are used.
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`own_namespace` | `bool`   | Include the namespace the agent is running in? | | no
+`names` | `[]string` | List of namespaces | | no
+
+### `selectors` block
+
+This contains optional label and field selectors to limit the discovery process to a subset of available resources.
+See [https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/](https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/)
+and [https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/) to learn more about the possible
+filters that can be used.
+
+Note: When making decision about using field/label selector make sure that this
+is the best approach - it will prevent Prometheus from reusing single list/watch
+for all scrape configs. This might result in a bigger load on the Kubernetes API,
+because per each selector combination there will be additional LIST/WATCH. On the other hand,
+if you just want to monitor small subset of pods in large cluster it's recommended to use selectors.
+The decision, to use selectors or not depends on the particular situation.
```

Comment:
I find this paragraph a little hard to follow. Can you try to rephrase this less conversationally?